core/templates/dev/head/domain/topic/SubtopicObjectFactory.js

```diff
@@ -50,35 +55,46 @@ oppia.factory('SubtopicObjectFactory', [
     };
 
     // Returns the skill ids in the subtopic.
-    Subtopic.prototype.getSkillIds = function() {
-      return this._skillIds;
+    Subtopic.prototype.getSkills = function() {
+      return this._skills.slice();
+    };
+
+    Subtopic.prototype.getIndexOfSkill = function(skillId) {
```

Comment:
Try not to have this. Refactor methods that use it to depend on stuff like skillId instead (or define a this.hasSkill(skillId) method, or similar).

As mentioned in another comment, index is pretty arbitrary and should ideally not be used as a distinguisher of identity. I think you'll only really need to use it in removeSkill() but keep that part of the processing local to that method. Everything else can be done with hasSkill().

Reply:
Done, removed this and added hasSkill().